odra-casper/livenet-env/src/livenet_contract_env.rs

```diff
@@ -0,0 +1,122 @@
+use blake2::digest::VariableOutput;
+use blake2::Blake2bVar;
+use odra_casper_client::casper_client::CasperClient;
+use odra_core::callstack::{Callstack, CallstackElement, Entrypoint};
+use odra_core::contract_register::ContractRegister;
+use odra_core::prelude::*;
+use odra_core::{Address, Bytes, OdraError, U512};
+use odra_core::{CallDef, ContractContext};
+use std::io::Write;
+use std::sync::{Arc, RwLock};
+
+pub struct LivenetContractEnv {
```

Comment:
I'm not sure but could `LivenetContractEnv` and `LivenetHostEnv` be the same struct (`LivenetEnv`) that implements both `ContractContext` and `HostContext`?

Reply:
Maybe @coderabbitai can create an issue for that?